shared/params.go

```diff
@@ -259,28 +263,23 @@ func ParseProductOrBrowserParams(r *http.Request) (products []Product, err error
 	return products, nil
 }
 
-// GetProductsForRequest parses the 'products' (and legacy 'browsers') params, returning
+// GetProductsOrDefault parses the 'products' (and legacy 'browsers') params, returning
 // the sorted list of products to include, or a default list.
-func GetProductsForRequest(r *http.Request) (products []Product, err error) {
-	products, err = ParseProductOrBrowserParams(r)
+func (filter TestRunFilter) GetProductsOrDefault() (products []Product) {
+	products = filter.Products
+	browserNames, err := GetBrowserNames()
```

Comment:
I know we've been doing this for a while, but I'm just wondering why we need to create a copy of the `shared.browserNames` slice every time?

Reply:
Only way to make it 'immutable'.